tests/components/hand-controls.test.js

```diff
@@ -0,0 +1,153 @@
+      process.nextTick(function () {
+        trackedControls = el.components['tracked-controls'];
+        // mock controller
+        trackedControls.controller = {id: 'Oculus Touch (Left)', connected: true};
+        // do the check
+        assert.ok(controllerComponent.isOculusTouchController());
+        done();
+      });
+    });
+
+    test('false if controller id does not start with "Oculus Touch"', function (done) {
+      var el = this.el;
+      var controllerComponent = el.components[controllerComponentName];
+      var trackedControls;
+      el.setAttribute('tracked-controls', '');
+      process.nextTick(function () {
+        trackedControls = el.components['tracked-controls'];
+        // mock controller
+        trackedControls.controller = {id: 'OpenVR Gamepad', connected: true};
+        // do the check
+        assert.notOk(controllerComponent.isOculusTouchController());
+        done();
+      });
+    });
+  });
+
+  suite('determineGesture', function () {
+    test('if nothing touched or triggered, no gesture', function (done) {
+      var el = this.el;
+      var controllerComponent = el.components[controllerComponentName];
+      // mock button / touch flags
+      controllerComponent.pressedButtons['grip'] = false;
+      controllerComponent.pressedButtons['trigger'] = false;
+      controllerComponent.pressedButtons['touchpad'] = false;
+      controllerComponent.pressedButtons['thumbstick'] = false;
+      controllerComponent.pressedButtons['menu'] = false;
+      controllerComponent.pressedButtons['AorX'] = false;
+      controllerComponent.pressedButtons['BorY'] = false;
+      controllerComponent.pressedButtons['surface'] = false;
+      // do the check
+      assert.notOk(controllerComponent.determineGesture());
+      process.nextTick(function () { done(); });
+    });
+
+    test('if non-Oculus Touch and only trackpad. pointing gesture', function (done) {
+      var el = this.el;
+      var controllerComponent = el.components[controllerComponentName];
+      var trackedControls;
+      el.setAttribute('tracked-controls', '');
+      process.nextTick(function () {
+        trackedControls = el.components['tracked-controls'];
+        // mock controller
+        trackedControls.controller = {id: 'Foobar', connected: true};
+        // mock button / touch flags
+        controllerComponent.pressedButtons['grip'] = false;
+        controllerComponent.pressedButtons['trigger'] = false;
+        controllerComponent.pressedButtons['trackpad'] = true;
+        controllerComponent.pressedButtons['thumbstick'] = false;
+        controllerComponent.pressedButtons['menu'] = false;
+        controllerComponent.pressedButtons['AorX'] = false;
+        controllerComponent.pressedButtons['BorY'] = false;
+        controllerComponent.pressedButtons['surface'] = false;
+        // do the check
+        assert.equal(controllerComponent.determineGesture(), 'pointing');
+        process.nextTick(function () { done(); });
```

Comment:
At this point you have already checked all your assertions and should be able to simply call `done()`.  I don't think `nextTick` is necessary. Can you try again?